code/screenshot.py

```diff
@@ -1,83 +1,79 @@
-    folder_path = screenshot_folder.get()
-    path = os.path.expanduser(os.path.join(folder_path, filename))
-    return os.path.normpath(path)
-
 
 @mod.action_class
 class Actions:
     def screenshot():
-        """takes a screenshot of the entire screen and saves it to the desktop as screenshot.png"""
-        img = screen.capture_rect(screen.main_screen().rect)
-        path = get_screenshot_path()
-
-        img.write_file(path)
-        app.notify(subtitle="Screenshot: %s" % path)
+        """Takes a screenshot of the entire screen and saves it to the pictures folder"""
+        screenshot_rect(screen.main_screen().rect)
 
     def screenshot_window():
-        """takes a screenshot of the current window and says it to the desktop as screenshot.png"""
-        img = screen.capture_rect(ui.active_window().rect)
-        path = get_screenshot_path()
-        img.write_file(path)
-        app.notify(subtitle="Screenshot: %s" % path)
+        """Takes a screenshot of the active window and saves it to the pictures folder"""
+        win = ui.active_window()
+        screenshot_rect(win.rect, win.app.name)
 
     def screenshot_selection():
-        """triggers an application is capable of taking a screenshot of a portion of the screen"""
-        command = screenshot_selection_command.get()
-        if command:
-            path = get_screenshot_path()
-            command = command.split()
-            command.append(path)
-            subprocess.Popen(command)
-            app.notify(subtitle="Screenshot: %s" % path)
-        else:
-            if active_platform == "windows":
-                actions.key("super-shift-s")
-            elif active_platform == "mac":
-                actions.key("ctrl-shift-cmd-4")
-            # linux is handled by the command by default
-            # elif active_platform == "linux":
+        """Triggers an application is capable of taking a screenshot of a portion of the screen"""
+        if app.platform == "windows":
+            actions.key("super-shift-s")
+        elif app.platform == "mac":
+            actions.key("ctrl-shift-cmd-4")
+        elif app.platform == "linux":
+            actions.key("shift-printscr")
 
     def screenshot_clipboard():
-        """takes a screenshot of the entire screen and saves it to the clipboard"""
-        img = screen.capture_rect(screen.main_screen().rect)
-        clip.set_image(img)
+        """Takes a screenshot of the entire screen and saves it to the clipboard"""
+        clipboard_rect(screen.main_screen().rect)
 
     def screenshot_window_clipboard():
-        """takes a screenshot of the window and saves it to the clipboard"""
-        img = screen.capture_rect(ui.active_window().rect)
-        clip.set_image(img)
+        """Takes a screenshot of the active window and saves it to the clipboard"""
+        clipboard_rect(ui.active_window().rect)
+
+
+def screenshot_rect(rect: ui.Rect, title: str = ""):
+    flash_rect(rect)
```

Comment:
I'm slightly hesitant on the flash at the moment due to this issue https://github.com/talonvoice/talon/issues/248, which would probably prevent the entire command from working

Reply:
Yes understand that. Probably should wait with this until that bug is fixed then. I only tested this on windows and linux and it's a great feature... if it's working of course.

Reply:
Wouldn't above issue already make it so that the capture doesn't work since the capture already relies upon ui.Rect? The flash just uses the same rect as the capture itself. Or is is just a problem with the canvas?

Reply:
You're correct-sorry about that.

Reply:
No problem :)